backend/lucene/src/main/java/org/hibernate/search/backend/lucene/types/sort/impl/BooleanFieldSortBuilderFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.lucene.types.sort.impl;
+
+import org.hibernate.search.backend.lucene.search.impl.LuceneSearchContext;
+import org.hibernate.search.backend.lucene.search.sort.impl.LuceneSearchSortBuilder;
+import org.hibernate.search.backend.lucene.types.converter.impl.BooleanFieldConverter;
+import org.hibernate.search.engine.search.sort.spi.FieldSortBuilder;
+
+public class BooleanFieldSortBuilderFactory extends AbstractStandardLuceneFieldSortBuilderFactory<Boolean> {
```

Comment:
Since you'll use `IntegerFieldSortBuilder`, you may as well remove this class and use `IntegerFieldSortBuilderFactory`. Just change `BooleanFieldSortBuilderFactory` by replacing `LuceneFieldConverter<Integer, ?>` (which is wrong) with `LuceneFieldConverter<?, Integer>`, and it will work just fine with your `BooleanFieldConverter`.